test/metabase/driver_test.clj

```diff
@@ -7,12 +7,26 @@
 (driver/register! ::test-driver, :abstract? true)
 
 (defmethod driver/supports? [::test-driver :foreign-keys] [_ _] true)
+(defmethod driver/database-supports? [::test-driver :foreign-keys] [_ _ db] (= db "dummy"))
+
 
 (deftest driver-supports?-test
   (is (= true
          (driver/supports? ::test-driver :foreign-keys)))
   (is (= false
-         (driver/supports? ::test-driver :expressions))))
+         (driver/supports? ::test-driver :expressions)))
+  (is (thrown? java.lang.Exception
+               (driver/supports? ::test-driver :some-made-up-thing))))
+
+(deftest database-supports?-test
+  (is (= true
+         (driver/database-supports? ::test-driver :foreign-keys "dummy")))
+  (is (= false
+         (driver/database-supports? ::test-driver :foreign-keys "not-dummy")))
+  (is (= false
+         (driver/database-supports? ::test-driver :expressions "dummy")))
```

Comment:
You can just do `(is (driver/database-supports? ...))` and `(is (not (driver/database-supports? ...)))` here. Sometimes the test tooling can give you more meaningful output when you're not using the `=` impl of `clojure.test/test-expr`